src/otx/algorithms/common/adapters/mmcv/configurer.py

```diff
@@ -413,6 +413,17 @@ def configure_hooks(
         if hasattr(cfg, "algo_backend"):
             self._update_caching_modules(cfg)
 
+        # Update adaptive repeat
+        if not self.training:
+            remove_from_configs_by_type(cfg.custom_hooks, "AdaptiveRepeatDataHook")
+        for custom_hook in cfg.custom_hooks:
+            if custom_hook["type"] == "AdaptiveRepeatDataHook":
+                data_cfg = cfg.get("data", {})
+                bs = data_cfg.get("train_dataloader", {}).get("samples_per_gpu", None)
+                bs = bs if bs is not None else data_cfg.get("samples_per_gpu", 0)
+                custom_hook["train_batch_size"] = bs
+                custom_hook["n_train_data"] = len(data_cfg.get("train", {}).get("otx_dataset", []))
+
```

Comment:
```suggestion
                custom_hook["n_train_data"] = len(data_cfg.get("train", {}).get("otx_dataset", []))
                break

```
How about to break to prevent it from iterating meaninglessly?